instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpServerAttributesExtractor.java

```diff
@@ -66,12 +74,14 @@ public void onStart(AttributesBuilder attributes, REQUEST request) {
   @Override
   public void onEnd(
       AttributesBuilder attributes,
+      Context context,
       REQUEST request,
       @Nullable RESPONSE response,
       @Nullable Throwable error) {
 
-    super.onEnd(attributes, request, response, error);
+    super.onEnd(attributes, context, request, response, error);
     set(attributes, SemanticAttributes.HTTP_SERVER_NAME, getter.serverName(request, response));
+    set(attributes, SemanticAttributes.HTTP_ROUTE, httpRouteHolderGetter.apply(context));
```

Comment:
I thought about adding a `route(context, request, response)` to the getter interface (with a default `HttpRouteHolder` implementation) but I decided against it: `HttpRouteHolder` not only stores the route, but also renames the server span whenever the route is updated; if we were to allow somebody to implement a `route(context, request, response)` method that overrode the default implementation (and skipped the `HttpRouteHolder` altogether), that instrumentation would not rename the server span after start -- and that would differ from how the `HttpRouteHolder` handles things.

In other words: using `route(request)` is fine, because the `HttpSpanNameExtractor` handles that, but if you want to set the route during/at the end of the request you have to use `HttpRouteHolder`.